### DISCUSSION TITLE
Pagination on data from a database using LIMIT and OFFSET

### DISCUSSION THREAD

**Discussion by uvinw** [windmill-labs/windmill#3184] [Questions]
I am fetching data from a Postgres query and rendering it into a table.

However, all the rows of the table are fetched to the client when the query is run, leading to a very unresponsive page.

Ideally I'd make use of LIMIT and OFFSET offered by Postgres to only bring in the required rows from the PG table to the client.

I could't find any examples in the documentation for something like this. Perhaps something like below would be ideal but I am unable to make the `Table` component supply the limit and offset values to my query.

Thanks! 

```
-- $1 limit
-- $2 offset
SELECT * from users LIMIT $1 OFFSET $2;
```

**Comment by rubenfiszel** [comment]
You are missing the type hints ::INT

![image](https://github.com/windmill-labs/windmill/assets/275584/95d83675-05fb-46ff-a392-f011817062c7)

**Reply by uvinw** [reply]
Thanks @rubenfiszel, for the suggestion.
I also connected the OFFSET to the `page` attribute of the table. However, LIMIT is set at a static 10.
The result is only 10 records are loaded into the table and the pagination is greyed out. Is there a table attribute I can connect the LIMIT to as well so I can keep navigating until the end of the dataset?
Thanks!

<img width="427" alt="image" src="https://github.com/windmill-labs/windmill/assets/12084652/4a01b803-7a2f-4317-9d7a-d76c37bab10f">

**Reply by uvinw** [reply]
@rubenfiszel any hints this? Thanks!
